### PR TITLE
init

### PR DESCRIPTION
**Pull request by smiller171**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#       }
+#     ]
+
+#     tags = merge(
+#       var.tags,
+#       { Name : var.name }
+#     )
+#   })
+#   filename = "${path.module}/test.yml"
+# }
```
Comment (sblack4) [inline main.tf:55]: if we don't need the commented out stuff then delete it?

Comment (smiller171) [reply main.tf:55]: You're absolutely right. Just forgot to clean up my testing code

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
+resource "aws_cloudformation_stack" "this" {
+  name               = var.name
+  on_failure         = "ROLLBACK"
+  timeout_in_minutes = var.timeout
```
Comment (cdaniluk) [inline main.tf:4]:
```suggestion
  timeout_in_minutes = var.cloudformation_timeout
```

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      }
+    ]
+
+    tags = merge(
+      var.tags,
+      { Name : var.name }
+    )
+  })
+
+  tags = merge(
```
Comment (cdaniluk) [inline main.tf:27]: move ahead of `template_body`

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  type        = string
+}
+
+variable "tags" {
+  default     = {}
+  description = "map of tags to use for CFN stack and component"
+  type        = map(string)
+}
+
+variable "timeout" {
```
Comment (cdaniluk) [inline variables.tf:61]:
```suggestion
variable "cloudformation_timeout" {
```

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        VolumeSize          = number
+        VolumeType          = string
+      })
+      NoDevice    = string
+      VirtualName = string
+    })
+  )
+}
+
+variable "components" {
```
Comment (cdaniluk) [inline variables.tf:22]:
```suggestion
variable "component_arns" {
```

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    block_device_mappings = var.block_device_mappings
+    description           = var.description
+    name                  = var.name
+    parent_image          = var.parent_image
+    platform              = var.platform
+    update                = var.update
+    version               = var.recipe_version
+
+    components = [
+      for component in var.components : {
```
Comment (cdaniluk) [inline main.tf:16]:
```suggestion
      for component in var.component_arns : {
```

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,71 @@
+variable "block_device_mappings" {
+  default     = null
+  description = "List of Maps of EBS volumes to mount https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-imagebuilder-imagerecipe-instanceblockdevicemapping.html"
+  type = list(
```
Comment (cdaniluk) [inline variables.tf:4]:
```suggestion

  type = list(
```

**Comment by smiller171** [comment]
> We don't have a standard for resource naming in CloudFormation, but all of our other code is CamelCased, not lazyCamelCased. Any reason not to change this?

You're confusing `camelCase` and `PascalCase` but I don't mind changing it since Amazon seems to prefer PascalCase in CFN. I'm just much more used to camelCase.

**Comment by cdaniluk** [comment]
> > We don't have a standard for resource naming in CloudFormation, but all of our other code is CamelCased, not lazyCamelCased. Any reason not to change this?
> 
> You're confusing `camelCase` and `PascalCase` but I don't mind changing it since Amazon seems to prefer PascalCase in CFN. I'm just much more used to camelCase.

I was a Java developer for 15 years and _everyone_ said CamelCase classes and lazyCamelCast objects. I don't care if we were wrong. After all, we were Java developers. Being wrong was what we did best.

**Comment by sblack4** [comment]
> We don't have a standard for resource naming in CloudFormation, but all of our other code is CamelCased, not lazyCamelCased. Any reason not to change this?

If we decide that here let's make a quick note of it like this: https://github.com/rhythmictech/engineering-standards/pull/3

Also, I checked the definitive answer for everything knowledge and I think we are talking about PascalCase: https://en.wikipedia.org/wiki/Camel_case

**Comment by cdaniluk** [comment]
> > We don't have a standard for resource naming in CloudFormation, but all of our other code is CamelCased, not lazyCamelCased. Any reason not to change this?
> 
> If we decide that here let's make a quick note of it like this: [rhythmictech/engineering-standards#3](https://github.com/rhythmictech/engineering-standards/pull/3)
> 
> Also, I checked the definitive answer for everything knowledge and I think we are talking about PascalCase: https://en.wikipedia.org/wiki/Camel_case

Yes, @smiller171 was 💯 and reminded me of how much happier I was back when I was writing Pascal. Long live Delphi, the original Monolith.

**Pull request merged**